envoy/access_log/access_log_config.h

```diff
@@ -70,11 +71,13 @@ template <class Context> class AccessLogInstanceFactoryBase : public Config::Typ
    * @param filter filter to determine whether a particular request should be logged. If no filter
    * was specified in the configuration, argument will be nullptr.
    * @param context access log context through which persistent resources can be accessed.
+   * @param command_parsers vector of command parsers that provide by the caller to be used for
+   * parsing custom substitution commands.
    */
-  virtual AccessLog::InstanceBaseSharedPtr<Context>
-  createAccessLogInstance(const Protobuf::Message& config,
-                          AccessLog::FilterBasePtr<Context>&& filter,
-                          Server::Configuration::FactoryContext& context) PURE;
+  virtual AccessLog::InstanceBaseSharedPtr<Context> createAccessLogInstance(
+      const Protobuf::Message& config, AccessLog::FilterBasePtr<Context>&& filter,
+      Server::Configuration::FactoryContext& context,
+      std::vector<Formatter::CommandParserBasePtr<Context>> command_parsers = {}) PURE;
```

Comment:
If possible, I think the 3rd param should either be a const-vector-opt-ref or an rvalue reference to imply that a move is needed. (I slightly prefer the 2nd option, but it depends on the where this vector is coming from).